### PR TITLE
Finalize theme token updates

### LINKED ISSUE #545: Finalize theme token updates
## Overview
As we move towards applying the new specs to our products, we need to update KDS to ensure the colors and fonts are available within the design system. The token to hex code mapping has been updated, but we will need to ensure that the components and mixin references are updated as well.

## To do
1. Reference marcella's working branch `test-themeing-updates` 
2. Audit and apply the proper colors referencing [the figma spec](https://www.figma.com/file/p7aOP2MBv0SED1uhaPwqEd/Product-rebrand?type=design&node-id=426-7545&mode=design&t=DTaXy2FwhcEAXTCl-0) - focusing in particular on where there are _updates to the token mapping for the component_ for example, where we used to be using `primary` (the purple) but now we are using a _secondary_ color token for the blues. Most of the interactive form elements are using _secondary_ tokens. (Buttons, input fields, etc.)
3. It may be easier to cross-reference with Kolibri, and you can use the new updates to run kolibri without using `yarn link`, using `yarn run devserver-with-kds /path/to/kds` instead

## Acceptance criteria 
- [ ] Standard `KComponents` have been upgraded to the updated theme colors including form elements, loaders, 

## Out of scope 
- edge cases in Kolibri or in Studio (i.e. any place there the tokens are referenced directly within a component, rather than just using the default `KComponent`, or where something like a hex code has snuck in)
- Font updates 
- Other related updates fixes in Kolibri or Studio (logos, icon colors, etc.)

### PR DESCRIPTION
**Pull request by LianaHarris360**
## Description
This PR updates the the brand token colors, names, and token mapping within components and mixin references and expands upon the initial token to hex code mapping work done by @marcellamaki 

#### Issue addressed
Addresses #545

### Before/after screenshots

| Before | After |
|-------| ------ |
| Colors Documentation Page: 
Screenshot 2024-02-16 at 12 12 17 PM Screenshot 2024-02-16 at 12 14 40 PM Screenshot 2024-02-16 at 12 16 13 PM Screenshot 2024-02-16 at 12 17 00 PM Screenshot 2024-02-16 at 12 27 12 PM | Screenshot 2024-02-16 at 12 13 25 PM Screenshot 2024-02-21 at 12 19 48 PM Screenshot 2024-02-21 at 12 19 31 PM Screenshot 2024-02-21 at 12 17 46 PM Screenshot 2024-02-16 at 12 32 44 PM
| KSwitch: KeenUiSelect, KeenUiSelectOption 
Screenshot 2024-02-13 at 1 42 01 PM Screenshot 2024-02-16 at 12 54 22 PM | Screenshot 2024-02-13 at 1 42 11 PM Screenshot 2024-02-16 at 1 12 09 PM
| KCheckbox 
Screenshot 2024-02-19 at 10 20 52 AM | Screenshot 2024-02-15 at 2 29 12 PM
| KTabs: KTabsList 
Screenshot 2024-02-19 at 10 22 58 AM | Screenshot 2024-02-14 at 2 27 03 PM
KButton: buttonMixin
 | Screenshot 2024-02-19 at 10 24 15 AM Screenshot 2024-02-19 at 10 24 22 AM| Screenshot 2024-02-19 at 10 26 51 AMScreenshot 2024-02-19 at 10 26 55 AM 1
| KDateRange: KDateDay, KDateCalendar 
Screenshot 2024-02-19 at 10 25 59 AM| Screenshot 2024-02-14 at 2 36 18 PM
| KExternalLink, KRouterLink
 Screenshot 2024-02-19 at 10 28 44 AM| Screenshot 2024-02-14 at 2 50 04 PM
| KRadioButton
 Screenshot 2024-02-19 at 10 29 47 AM| Screenshot 2024-02-14 at 2 53 21 PM
| KTextBox 
 Screenshot 2024-02-19 at 10 33 01 AM | Screenshot 2024-02-15 at 3 08 13 PM
| KLinearLoader
Screenshot 2024-02-21 at 12 10 57 PM | Screenshot 2024-02-21 at 12 11 11 PM
| KCircularLoader
Screenshot 2024-02-21 at 12 13 34 PM | Screenshot 2024-02-21 at 12 13 26 PM

## Changelog

- [#551]
  - **Description:** Updates `brand` colors, `palette` colors, and `token`s.
    - Breaking changes:
      - Removed `palette` colors: `purple`, `deeppurple`, `indigo`, `brown`, `cyan`, `teal`, `lightgreen`, `lime`, `amber`, `deeporange`, `bluegrey`
      -  Removed `palette.grey` scales: `v_300`, `v_500`, `v_700`, `v_900`
      - Removed `brand` and `palette` scales (except `palette.grey`): `v_50`,`v_100`, `v_300`, `v_500`, `v_700`, `v_900`
      - Removed content-related tokens: `exercise`, `video`, `audio`, `document`, `html5`, `slideshow`
      - Removed other tokens: `appBarFullscreen`, `appBarFullscreenDark`, `linkDark`
     - Other changes:
       - Some existing `palette` colors look differently
       - Adds new tokens and palette
       - Global styles: `<body>` background color changed from `grey.v_100` to lighter `grey.v_50`.
  - **Products impact:** new API, updated API, deleted API
  - **Addresses:** https://github.com/learningequality/kolibri-design-system/issues/545
  - **Components:**
  - **Breaking:** yes
  - **Impacts a11y:** no
  - **Guidance:** Address all breaking changes by searching for removed palette colors, scales, and tokens. Study the updated "Colors" KDS documentation page and replace them by relevant colors/scales/tokens. Also search for any hardcoded hex,rgb(a),hsl(a), or named colors (such as 'white') and theme them instead. Visually check places that use existing palette and adjust scale (you may need to increase it as many colors are lighter). You may also see if there are any minor useful updates to in regards to new tokens (e.g. replacing a `palette` color with a new `token` that describes function of the color better). If you use `generateGlobalStyles` that generates background color for `<body>` and use `grey.v_100` in some components to match the background color, you may need to update it to `grey.v_50`.
     
[#551]: https://github.com/learningequality/kolibri-design-system/pull/551

## (optional) Implementation notes
- This line was removed from the KDS Colors Documentation Page because the mentioned colors were removed: 

  > The Kolibri Design System has what we call "Kolibri Purple" as its primary and "Kolibri Teal" as its secondary, available across the full brightness scale:

- KContentRenderer was removed from the table of contents because it displayed an error page

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] The change is described in the changelog section above

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?

## After review

- [ ] The changelog item has been pasted to the `CHANGELOG.md`

### CODE REVIEW

**Comment by MisRob** [comment]
Thanks @LianaHarris360.

Redirecting here the conversation about the changelog from the closed PR:

---

Thank you, new updates make sense. I see one place that could deserve more detail in my view

When you say

> new content-related token labels have been introduced and existing ones have been renamed

Could you list new labels and before/after of renamed token labels here? So we can find them in the products easily and rename them without the need to dig into this PR's diff. These are changes in public API and we will be specific in the release notes (this is different from just tweaking an existing color which we don't need to document in such detail.

> I was planning to update the changelog with more info but wasn't too sure about the level of detail to include

Generally, you could just imagine that you're someone who's not familiar with the code diff in detail and tried to introduce new theming to Kolibri/Studio/KDP. Anything that would be useful for you to know will serve as good guidance :)

Thank you. This will be helpful later on as we are updating Kolibri, Studio, and KDP.

**Comment by MisRob** [comment]
FYI I'm gradually reviewing this as I'm working on rebranding Kolibri, so it may take a while to post final feedback, but it's in progress.

**Review by marcellamaki — COMMENTED** [review]
This looks great, @LianaHarris360, thank you so much for the quick and thorough work on this. 

I spoke briefly with @MisRob and I think for the rest of the week, perhaps we can keep this open in case there are any changes to be made here that are discovered in the updates to Kolibri and Studio, and they can just be made directly here, so I am commenting rather than approving. Managing this will take some working cooperatively and communication, but I think we can do it. :) 

Overall though, great start. For those who will be doing the updates in our products, I think this is ready to work from at least to do some next steps. Please let me know if you have any questions, either here or in the slack thread in #product.

**Comment by MisRob** [comment]
Hi, I did first review and overall great work @marcellamaki @LianaHarris360, thank you. See notes below from the review and also my parallel work of using it in Kolibri. It's work in progress so it's possible that I will post some more over time, but I think this could be all when it comes to high-level:

(1) I located more breaking changes that are not in the changelog yet. @LianaHarris360 I needed to note them down for myself when upgrading Kolibri, so I hope it's fine if I just updated the changelog directly rather than referencing it.

**(2) One high-level issue that's blocking some Kolibri work is removal of yellow color from the full palette.** The only way how to use yellow now is via `brand.secondary`. That's not suitable in all cases because for example for warning yellow icon, we don't want to use `brand.secondary`. It's not only about color, but also context. Practically, in another theme that would use green as secondary color, the warning icon would end up green, which is not desired. `brand.secondary` need to be used only in places that have something to do with brand (e.g. buttons and navbars). In another places, we will need `palette.yellow` even though it's the same color as `brand.secondary` under the hood. I'd suggest we return yellow to the palette.  

**(3) Relatedly to (2), I think that except of tokens that are related to brand (such as tokens.primary, tokens.primaryDark, ...), tokens need to be created via palette rather than brand colors.** For example, `tokens.practice` need to be created via some `palette` blue color rather than `brand`.

Screenshot from 2024-02-21 10-10-18

Here, I only pointed to a few places I looked into in more detail, but I think there are some more. @LianaHarris360 could you please revisit it as a whole and make sure that there's distinction between blue and yellow for brand and palette?

**Comment by MisRob** [comment]
Also in regards to (2), the yellow we need in the palette is this one. I think that's the same yellow as the one used for brand secondary, but please double-check

https://www.figma.com/file/p7aOP2MBv0SED1uhaPwqEd/Product-rebrand?node-id=766%3A13416&mode=dev

Screenshot from 2024-02-21 10-21-02

Note that Figma has mastered and warning `token`s for this yellow. No problem to keep using `palette.yellow` as soon as it's back in the palette if this is not in scope of this work, but it'd be good to follow up to map all figma tokens at some point.

**Comment by MisRob** [comment]
Also in regards to (2), the yellow we need in the palette is this one. I think that's the same yellow as the one used for brand secondary, but please double-check

https://www.figma.com/file/p7aOP2MBv0SED1uhaPwqEd/Product-rebrand?node-id=766%3A13416&mode=dev

Screenshot from 2024-02-21 10-21-02

Note that Figma has `token.warning` for this yellow. No problem to use `palette.yellow` as soon as it's in the palette if `warning` token is not in scope of this work, but it'd be good to follow up to map all figma tokens at some point.

Here's one of the relevant Kolibri places https://github.com/learningequality/kolibri/blob/release-v0.16.x/kolibri/plugins/device/assets/src/views/DeprecationWarningBanner.vue#L10-L14

**Comment by MisRob** [comment]
Actually, let's put both yellow and blue colors that are used for brand into palette. I think it's communicated in the designs as well (even though it's not duplicated in Other Colors). E.g. see that brand.primary.v_200 is based on blue_200, and similar.

Screenshot from 2024-02-21 13-06-27

**Comment by MisRob** [comment]
Actually, let's put both yellow and blue colors that are used for brand into palette. I just noticed it's communicated in the designs as well (even though it's not duplicated in Other colors so it was a bit confusing perhaps). E.g. see that brand.primary.v_200 is based on blue_200, and similar.

Screenshot from 2024-02-21 13-06-27

File: lib/styles/colorsDefault.js [hunk 1]
```diff
@@ -1,70 +1,66 @@
...
-  appBarFullscreenDark: 'black',
-  link: 'brand.primary.v_400',
-  linkDark: 'brand.primary.v_700',
-  loading: 'brand.secondary.v_200',
-  focusOutline: 'brand.secondary.v_200',
+  primary: 'brand.primary.v_1000',
+  primaryDark: 'brand.primary.v_1100',
+  secondary: 'brand.secondary.v_1000',
+  secondaryDark: 'brand.secondary.v_1100',
+  appBar: 'brand.secondary.v_800',
```
Comment (MisRob) [inline lib/styles/colorsDefault.js:7]: Could we please document all tokens on Colors page such as `appBar` and possibly others that are missing? I can't find it at https://deploy-preview-551--kolibri-design-system.netlify.app/colors/# It seems some of them were forgotten before, it'd be helpful to include them now as we're upgrading though.

**Comment by MisRob** [comment]
I'd like to ask for updating https://github.com/learningequality/kolibri-design-system/blob/bfedccc2836003be915cb9be6a5eb8c5ba0fefce/lib/styles/generateGlobalStyles.js#L31 to `grey.v_50`.

This will change the darker gray area in the bottom half in Kolibri to match the new lighter upper part.

Screenshot from 2024-02-21 14-07-09

**Comment by LianaHarris360** [comment]
If it's easier for you to update the changelog directly, I don't mind at all! I will go over the changes, document the missing colors, and update this PR based on your review notes.

**Comment by MisRob** [comment]
Thank you, I updated it in the description. Yes, it would be nice to keep it up to date as we're doing more changes here.

**Comment by MisRob** [comment]
New changes are looking fine, thank you @LianaHarris360

**Review by MisRob — APPROVED** [review]
Thank you @LianaHarris360, great work. The newest additions are looking fine to me and I think we resolved all outstanding issues. I will merge in preparation for the release. If there's some other work needed, we can have a new PR.

**Pull request merged**